internal/generator/kzg/template/marshal.go.tmpl

```diff
@@ -4,15 +4,62 @@ import (
 	"github.com/consensys/gnark-crypto/ecc/{{ .Name }}"
 )
 
-// WriteTo writes binary encoding of the SRS
+// WriteTo writes binary encoding of the ProvingKey
+func (pk *ProvingKey) WriteTo(w io.Writer) (int64, error) {
+	// encode the ProvingKey
+	enc := {{ .CurvePackage }}.NewEncoder(w)
+
+	toEncode := []interface{}{
+		pk.G1,
+	}
+
+	for _, v := range toEncode {
+		if err := enc.Encode(v); err != nil {
+			return enc.BytesWritten(), err
+		}
+	}
+
+	return enc.BytesWritten(), nil
+}
+
+// WriteRawTo writes binary encoding of Proof to w without point compression
+func (vk *VerifyingKey) WriteRawTo(w io.Writer) (int64, error) {
+	return vk.writeTo(w, {{.CurvePackage}}.RawEncoding())
+}
+
+// WriteTo writes binary encoding of the VerifyingKey
+func (vk *VerifyingKey) WriteTo(w io.Writer) (int64, error) {
+	return vk.writeTo(w)
+}
+
+func (vk *VerifyingKey) writeTo(w io.Writer, options ...func(*{{.CurvePackage}}.Encoder)) (int64, error) {
+	// encode the VerifyingKey
+	enc := {{ .CurvePackage }}.NewEncoder(w, options...)
+
+	toEncode := []interface{}{
+		&vk.G2[0],
+		&vk.G2[1],
+		&vk.G1,
+	}
+
+	for _, v := range toEncode {
+		if err := enc.Encode(v); err != nil {
+			return enc.BytesWritten(), err
+		}
+	}
+
+	return enc.BytesWritten(), nil
+}
+
+// WriteTo writes binary encoding of the entire SRS
 func (srs *SRS) WriteTo(w io.Writer) (int64, error) {
-	// encode the SRS
+	// encode the VerifyingKey
```

Comment:
bad comment.

Here I would rather call srs.Pk.WriteTo, srs.Vk.WriteTo, easier to maintain for future changes.

There is a duplicate point but it's not a crazy overhead.